PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
+      sourceApplication: options[UIApplicationOpenURLOptionsKey.sourceApplication] as? String,
+      annotation: [: ]
+    )
   }
-  func applicationWillResignActive(application: UIApplication) {}
-  func applicationDidEnterBackground(application: UIApplication) {}
-  func applicationWillEnterForeground(application: UIApplication) {}
-  func applicationDidBecomeActive(application: UIApplication) {}
-  func applicationWillTerminate(application: UIApplication) {}
-  
-  func signIn(signIn: GIDSignIn!, didSignInForUser user: GIDGoogleUser!,
-              withError error: NSError!) {
+  func applicationWillResignActive(_ application: UIApplication) {}
+  func applicationDidEnterBackground(_ application: UIApplication) {}
+  func applicationWillEnterForeground(_ application: UIApplication) {}
+  func applicationDidBecomeActive(_ application: UIApplication) {}
+  func applicationWillTerminate(_ application: UIApplication) {}
+
+  func sign(_ signIn: GIDSignIn, didSignInFor user: GIDGoogleUser, withError error: Error?) {
     if let error = error {
       print("signIn error : \(error.localizedDescription)")
       return
     }
-    
-    if (self.user == nil) {
+
+    if self.user == nil {
       self.user = user
-      let authentication = user.authentication
-      let credential =
-        FIRGoogleAuthProvider.credentialWithIDToken(
-          authentication.idToken,
-          accessToken: authentication.accessToken)
-      FIRAuth.auth()?.signInWithCredential(credential) { (user, error) in
-        print("Signed-in to Firebase as \(user!.displayName!)")
+      guard let authentication = user.authentication else { return }
+      let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                                                     accessToken: authentication.accessToken)
+      Auth.auth().signInAndRetrieveData(with: credential) { (authResult, error) in
+          if error != nil {
+              return
+          }
+        print("Signed-in to Firebase as \(String(describing: authResult?.user.displayName))")
         let nav = UINavigationController(
           rootViewController: self.tabBarController!)
-        self.window?.rootViewController?.presentViewController(
-          nav,
-          animated: true, completion: nil)
-        self.ref = FIRDatabase.database().reference()
+        self.window?.rootViewController?.present( nav, animated: true, completion: nil)
+        self.ref = Database.database().reference()
         self.inbox = "client-" + String(abs(self.user.userID.hash))
 
         self.msgViewController!.inbox = self.inbox
         self.msgViewController!.ref = self.ref
-        
+
         self.requestLogger()
       }
     }
   }
-  
-  func signOut(sender:UIButton) {
+
+  func sign(_ signIn: GIDSignIn, didDisconnectWith user: GIDGoogleUser, withError error: Error) {
+    // Perform any operations when the user disconnects from the app.
+  }
+
+  @objc
+  func signOut(_ sender: UIButton) {
```

Comment:
We don't have a style guide so you can ignore this comment if you'd prefer, but our other Swift samples inline `@objc` annotations.

Reply:
Done.